include/zephyr/net/coap.h

```diff
@@ -841,8 +919,10 @@ bool coap_register_observer(struct coap_resource *resource,
  *
  * @param resource Resource in which to remove the observer
  * @param observer Observer to be removed
+ *
+ * @return true if the observer was found and removed.
  */
-void coap_remove_observer(struct coap_resource *resource,
+bool coap_remove_observer(struct coap_resource *resource,
```

Comment:
BTW, we are changing public APIs here, we probably need to put something to migration notes about these changes.

Reply:
Yes, is changing the return type from `void` to `bool` that intrusive? I guess its an issue if you have compiler warnings/error for unused return values.

Reply:
Well, there is a compile error at least. It would just be great if any public API changes are documented somewhere and the migration doc seems to be a good place for those.

Reply:
I've added an item to the migration guide.